homeassistant/components/assist_satellite/entity.py

```diff
@@ -36,11 +46,112 @@ class AssistSatelliteEntity(entity.Entity):
     entity_description: AssistSatelliteEntityDescription
     _attr_should_poll = False
     _attr_state: AssistSatelliteState | None = None
+    _attr_supported_features = AssistSatelliteEntityFeature(0)
 
     _conversation_id: str | None = None
     _conversation_id_time: float | None = None
 
-    _run_has_tts: bool = False
+    _is_announcing: bool = False
+    _tts_finished_event: asyncio.Event | None = None
+    _wake_word_future: asyncio.Future[str | None] | None = None
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        self._set_state(AssistSatelliteState.LISTENING_WAKE_WORD)
+
+    @property
+    def is_announcing(self) -> bool:
+        """Returns true if currently announcing."""
+        return self._is_announcing
+
+    async def async_announce(
+        self,
+        announce_text: str,
+        announce_media_id: str | None = None,
+        pipeline_entity_id: str | None = None,
```

Comment:
I believe that we should move to make `pipeline_entity_id` and the other one properties on the entity instead of passing them around.